torch/jit/__init__.py

```diff
@@ -1402,6 +1403,14 @@ def _unwrap_optional(x):
     return x
 
 
+def _unwrap_tuple(x):
```

Comment:
hmm do we need to think more about this? I think we want to ultimately remove any those unwrap functions if possible, adding more and more might result in a non-easy revert in the future. At least we should think how to provide user a API or something they don't need to care about.

Reply:
If we want to support it at the language level instead of having these hacks I would imagine it would look something like

```python
@torch.jit.script
def fn(x):
    # type: (Union[int, float]) -> Union[int, float]
    if isinstance(x, int):
        return 3
    else:
        return 3.5
```

And similarly to #14533 we would only emit the branches for the types seen at compile time

Reply:
Hmmm It's hard to control those meta programming conditions since it might be a runtime known value, it might be ok for us as a temp hack to add unwrap/wrap tuples in supporting lstm modules, I guess we just need to be aware to not stacking too much on them